### ISSUE TITLE
Problem understanding the progress of evaluation cost.

### ISSUE THREAD

**Issue by recursive-beast** [TimotheeChauvin/NNDL-solutions#4]
Hello, I just finished reading the 3rd chapter from nielson's book, and i tried to compare between training the neural network using quadratic cost vs using the cross entropy cost , and the training cost decreases as expected ( I'm using L2 regularization ) , but i noticed in the plots that the evaluation cost increases over epochs instead of decreasing, while the accuracy of both training and evaluation increases as expected .

can you explain to me why is this happening please ?

here are the plots I'm talking about :

![image](https://user-images.githubusercontent.com/41545690/76239885-95d6f180-6232-11ea-9ca8-1430180aa233.png)

and here's how I'm calculating the [regularized](http://neuralnetworksanddeeplearning.com/chap3.html#regularization) total cost of each data set ( training and evaluation) : 

```python
def total_cost(self, data: Loader, lmbda: float):
    """Return the total cost for the data set ``data``"""

    total_cost = 0.0
    data_size = 0

    for x, y in data:
        data_size += 1
        a = self.feedforward(x)

        if type(y) == int:
            y = unitVector(y)

        total_cost += self.cost(a, y)

    total_cost += 0.5 * lmbda * sum(np.linalg.norm(w) ** 2 for w in self.weights)

    return total_cost / data_size

```
